ui/app/models/pki/certificate/base.js

```diff
@@ -43,33 +43,59 @@ export default class PkiCertificateBaseModel extends Model {
       'The time after which this certificate will no longer be valid. This can be a TTL (a range of time from now) or a specific date.',
     editType: 'yield',
   })
-  customTtl; // combines ttl and notAfter into one input <PkiNotValidAfterForm>
+  customTtl; // sets ttl and notAfter via one input <PkiNotValidAfterForm>
+
+  @attr('boolean', {
+    label: 'Exclude common name from SANs',
+    subText:
+      'If checked, the common name will not be included in DNS or Email Subject Alternate Names. This is useful if the CN is a human-readable identifier, not a hostname or email address.',
+    defaultValue: false,
+  })
+  excludeCnFromSans;
+
+  @attr('string', {
+    label: 'Subject Alternative Names (SANs)',
+    subText:
+      'The requested Subject Alternative Names; if email protection is enabled for the role, this may contain email addresses. Add one per row.',
+    editType: 'stringArray',
+  })
+  altNames;
+
+  // SANs below are editType: stringArray from openApi
+  @attr('string', {
+    label: 'IP Subject Alternative Names (IP SANs)',
+    subText: 'Only valid if the role allows IP SANs (which is the default). Add one per row.',
+  })
+  ipSans;
+
+  @attr('string', {
+    label: 'URI Subject Alternative Names (URI SANs)',
+    subText:
+      'If any requested URIs do not match role policy, the entire request will be denied. Add one per row.',
+  })
+  uriSans;
+
+  @attr('string', {
+    subText:
+      'Requested other SANs with the format <oid>;UTF8:<utf8 string value> for each entry. Add one per row.',
+  })
+  otherSans;
 
   // Attrs that come back from API POST request
-  @attr({ masked: true, label: 'CA Chain' }) caChain;
+  @attr({ label: 'CA Chain', masked: true }) caChain;
   @attr('string', { masked: true }) certificate;
   @attr('number') expiration;
-  @attr('number', { formatDate: true }) revocationTime;
   @attr('string', { label: 'Issuing CA', masked: true }) issuingCa;
-  @attr('string') privateKey;
-  @attr('string') privateKeyType;
+  @attr('string') privateKey; // only returned for type=exported
+  @attr('string') privateKeyType; // only returned for type=exported
+  @attr('number', { formatDate: true }) revocationTime;
   @attr('string') serialNumber;
 
-  // Parsed from cert in serializer
-  @attr('number', { formatDate: true }) notValidAfter;
-  @attr('number', { formatDate: true }) notValidBefore;
-  @attr('string', { label: 'URI Subject Alternative Names (URI SANs)' }) uriSans;
```

Comment:
these sans are readonly for issuers, but edit-able for certificates moved up and added clarifying subtext